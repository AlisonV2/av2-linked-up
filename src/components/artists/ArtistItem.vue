<template>
  <div class="card artist-card">
    <div class="card-overlay"></div>
    <div class="card-details fade-in-top text-center">
      <h5 class="card-title">{{ artist.name }}</h5>
      <p class="card-text">{{ artist.shop }}</p>
      <p class="card-text">{{ artist.city }} ({{ artist.zip }})</p>
    </div>
    <img
      :src="artist.thumbnail"
      class="card-img-top"
      :alt="artist.name"
    />
    <div class="card-body">
      <h5 class="card-title">{{ artist.name }}</h5>
    </div>
  </div>
</template>

<script>
/**
 * @type Component
 * @name ArtistItem
 * @vue-prop {object} artists
 */
export default {
  name: 'ArtistItem',
  props: ['artist'],
};
</script>

<style lang="scss" scoped>
.artist-card {
  .card-title {
    font-family: $accent-font;
    color: $primary;
    font-size: 1.2rem;
  }
  .card-text {
    color: $light;
  }
  .card-details {
    position: absolute;
    text-align: center;
    padding-left: 1em;
    padding-right: 1em;
    width: 100%;
    top: 50%;
    left: 50%;
    opacity: 0;
    transform: translate(-50%, -50%);
    transition: all 0.3s ease-in-out 0s;
    .card-title {
      margin-bottom: 2rem;
    }
  }
  .card-overlay {
    background: rgba(0, 0, 0, 0.8);
    position: absolute;
    height: 100%;
    width: 100%;
    left: 0;
    top: 0;
    bottom: 0;
    right: 0;
    opacity: 0;
    transition: all 0.4s ease-in-out 0s;
  }
}

.artist-card:hover {
  .card-details {
    display: block;
  }
  .card-overlay {
    opacity: 1;
  }
  .card-details {
    top: 50%;
    left: 50%;
    opacity: 1;
  }
}

.fade-in-top {top: 20%};
</style>
