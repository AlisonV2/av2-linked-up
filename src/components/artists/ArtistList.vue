<template>
  <div class="row">
    <div
      class="col-8 col-md-6 col-lg-4 col-xl-3 artists-list"
      v-for="artist in artists"
      :key="artist.name"
    >
      <router-link :to="{ name: 'Profile', params: { id: artist.id } }">
        <ArtistItem :artist="artist" />
      </router-link>
    </div>
  </div>
</template>

<script>
import ArtistItem from '@/components/artists/ArtistItem';
import * as Sentry from '@sentry/vue';

/**
 * @exports ArtistList
 * @type {Component}
 * @vue-data {array} artists
 * @vue-event created - Dispatch store action
 */
export default {
  name: 'ArtistList',
  components: {
    ArtistItem,
  },
  data() {
    return {
      artists: [],
    };
  },
  /**
   * @description Dispatch getAllArtists action on created hook
   * Get results from store
   * @method created
   * @returns {array}
   * @async
   */
  async created() {
    await this.$store.dispatch('getAllArtists')
    .then(() => {
      this.artists = this.$store.getters.getAllArtists;
    })
    .catch((err) => Sentry.captureException(err));
  },
};
</script>
