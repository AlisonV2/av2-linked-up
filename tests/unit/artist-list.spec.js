import { shallowMount, RouterLinkStub } from '@vue/test-utils';
import { createStore } from 'vuex';
import ArtistList from '@/components/artists/ArtistList';
import ArtistItem from '@/components/artists/ArtistItem';

const actions = {
    getAllArtists: jest.fn(),
}
const store = createStore({
  state: {
    artists: [
      {
        id: 'mR9w90C9fMQfVHJWjGo1TrBxPuk1',
        name: 'Dirty Jonz',
        shop: 'Le Dernier Rituel',
        city: 'BÃ¨gles',
        zip: '33130',
      },
    ],
  },
  actions,
  getters: {
    getAllArtists(state) {
      return state.artists;
    },
  },
});

/**
 * @module ArtistListTest
 */
describe('ArtistList.vue', () => {
  it('Tests if dispatch action is called on created', () => {
    const component = shallowMount(ArtistList, {
      data() {
        return { artists: [] };
      },
      global: {
        plugins: [store],
        components: {
          'router-link': RouterLinkStub,
          ArtistItem: ArtistItem,
        },
      },
    });
    ArtistList.created.call(component.vm);
    expect(actions.getAllArtists).toHaveBeenCalled();
  });
});
