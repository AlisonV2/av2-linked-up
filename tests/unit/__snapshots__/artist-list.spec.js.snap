// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArtistList.vue ArtistList Snapshot 1`] = `
<div
  class="row"
>
  
  
</div>
`;
